regolith/helpers/u_contacthelper.py

```diff
@@ -71,34 +73,39 @@ def construct_global_ctx(self):
     def db_updater(self):
         rc = self.rc
         name = HumanName(rc.name)
-        if not rc.date:
-            now = dt.datetime.now()
-        else:
-            now = date_parser.parse(rc.date).date()
-        if not rc.id:
-            key = str(name.first[0].lower().replace(" ", "") + name.last.lower().replace(" ", ""))
-        else:
-            key = rc.id
-        filterid = {'_id': key}
-        coll = self.gtx[rc.coll]
-        pdoc = {'_id': key}
-        pdocl = list(filter(lambda doc: doc["_id"] == key, coll))
-        if len(pdocl) > 0:
-            current = rc.client.find_one(rc.database, rc.coll, filterid)
-            notes = current.get('notes', [])
-            aliases = current.get('aka', [])
-        else:
+        found_contacts = fragment_retrieval(self.gtx['contacts'], ["_id", "aka", "name"], rc.name)
+        found_contacts.sort(key=lambda x: x['_id'], reverse=False)
+        index = list(range(2, (len(found_contacts) + 2)))
+        if not rc.number:
+            print("Please rerun the helper specifying '-n list-index' to update item number 'list-index':")
+            print(f"{1}. {rc.name} as a new contact")
+            for i, j in zip(index, found_contacts):
+                print(f"{i}. {j['name']}    id: {j['_id']}")
+            return
+        pdoc = {}
+        if int(rc.number) == 1:
             if not rc.institution:
-                raise RuntimeError("Institution is required to create a new contact")
+                raise RuntimeError("institution is required to create a new contact")
+            if not rc.id:
+                key = str(name.first[0].lower().replace(" ", "") + name.last.lower().replace(" ", ""))
+            else:
+                key = rc.id
+            pdoc.update({"name": name.full_name})
             pdoc.update({"date": dt.date.today()})
             pdoc.update({"institution": rc.institution})
             notes = []
             aliases = []
             uniqueidentifier = str(uuid.uuid4())
             pdoc.update({'uuid': uniqueidentifier})
-        #FIXME
-        # if rc.e_email:
-        #     pdoc.update({"email": rc.email})
+        else:
+            current = found_contacts[int(rc.number)-2]
```

Comment:
I would type-check rc.number right at the top and throw an exception if it is not an int.  then remove the int here. Otherwise the user may get unexpected behavior (no magic rule)

Reply:
I added type=int to the subparser,  and I think it will make sure that it is an integer!
I did a test passing a date and it yelled at me.

Reply:
nice work!